### PR TITLE
update quote parser

### PR DESCRIPTION
**Pull request by siddontang**
fix `select '\'a\'';`

### CODE REVIEW

**Comment by siddontang** [comment]
@shenli

File: parser/scanner.l [hunk 1]
```diff
@@ -492,9 +493,26 @@ sys_var		"@@"(({global}".")|({session}".")|{local}".")?{ident}
 
 \"			l.sc = S1
 '			l.sc = S2
-
-<S1>(\\.|[^\"]|\"\")*\"		return l.str(lval, "\"")
-<S2>((\\')|[^']|\n|'')*'	return l.str(lval, "'")
+`			l.sc = S4
+
+<S1>[^\"\\]*		l.stringLit = append(l.stringLit, l.val...)	
+<S1>\\.			l.stringLit = append(l.stringLit, l.val...)
```
Comment (shenli) [inline parser/scanner.l]: Why not merge this rule to the rule in the above line?

Comment (siddontang) [inline parser/scanner.l]: above line handles not " and \\, this line handles \\ followed by any character, they are not same, and I think distinguish them is better to know.

**Comment by shenli** [comment]
LGTM

File: parser/scanner.l [hunk 2]
```diff
@@ -914,16 +929,14 @@ func (l *lexer) npos() (line, col int) {
 
 func (l *lexer) str(lval *yySymType, pref string) int {
 	l.sc = 0
-	s := string(l.val)
 	// TODO: performance issue.
+	s := string(l.stringLit)
+	l.stringLit = l.stringLit[0:0]
```
Comment (qiuyesuifeng) [inline parser/scanner.l]: Here why use l.stringLit[0:0]?

Comment (siddontang) [inline parser/scanner.l]: clear and for next use.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**